### DISCUSSION TITLE
How to filter base on multiple meta fields?

### DISCUSSION THREAD

**Discussion by Temepest74** [spatie/laravel-permission#1844] [Q&A] [answered]
On the plugin page, I saw how to filter based on one metadata field, but how can I do it on two or more fields?
Currently I am doing this on the already generated collection, but this approach is slower

**Comment by erikn69** [comment, marked as answer]
I don't know what means metadata for you, and there is no results for metadata on docs, but on params you can set all the filters that you need

https://github.com/spatie/laravel-permission/blob/58d5eb6c7b0eafa8bdf0a93d1ca5c214a05344cb/src/Models/Permission.php#L146-L151

Example
```php
$filtered = Permission::getPermissions(['name' => 'your_name', 'guard_name' => 'web', 'type' => 'Main']);
```

**Reply by Temepest74** [reply]
I realized i posed this on the wrong repository:))) sryyy
